src/sentry_slice.c

```diff
@@ -13,3 +23,72 @@ sentry__slice_eq(sentry_slice_t a, sentry_slice_t b)
 {
     return a.len == b.len && memcmp(a.ptr, b.ptr, a.len) == 0;
 }
+
+sentry_slice_t
+sentry__slice_split_at(sentry_slice_t a, char c)
+{
+    for (size_t i = 0; i < a.len; i++) {
+        if (a.ptr[i] == c) {
+            a.len = i;
+            return a;
+        }
+    }
+    return a;
+}
+
+size_t
+sentry__slice_find(sentry_slice_t a, char c)
+{
+    for (size_t i = 0; i < a.len; i++) {
+        if (a.ptr[i] == c) {
+            return i;
+        }
+    }
+    return (size_t)-1;
+}
+
+static bool
+is_space(char c)
+{
+    switch (c) {
+    case ' ':
+    case '\t':
+    case '\r':
+    case '\n':
+        return true;
+    default:
+        return false;
+    }
+}
+
+sentry_slice_t
+sentry__slice_trim(sentry_slice_t a)
+{
+    while (a.len > 0 && is_space(a.ptr[0])) {
+        a.ptr++;
+        a.len--;
+    }
+    while (a.len > 0 && is_space(a.ptr[a.len - 1])) {
+        a.len--;
+    }
+
+    return a;
+}
+
+bool
+sentry__slice_pop_uint64(sentry_slice_t *a, uint64_t *num_out)
+{
+    char *buf = sentry_malloc(a->len + 1);
```

Comment:
you have to free this. Also, you do need the copy to null-terminate it for `strtoll`?

Reply:
yeah :(